packages/core/src/services/documentVersions/create.ts

```diff
@@ -1,25 +1,28 @@
 import {
+  database,
   DocumentVersion,
   documentVersions,
   findCommit,
   Result,
   Transaction,
   type DocumentType,
 } from '@latitude-data/core'
+import { NotFoundError } from '$core/lib/errors'
 
-import createCommit from '../commits/create'
-
-function createDocument({
-  name,
-  commitId,
-  parentId,
-  documentType,
-}: {
-  name: string
-  commitId: number
-  parentId?: number
-  documentType?: DocumentType
-}) {
+function createDocument(
+  {
+    name,
+    commitId,
+    parentId,
+    documentType,
+  }: {
+    name: string
+    commitId: number
+    parentId?: number
+    documentType?: DocumentType
+  },
+  db = database,
```

Comment:
as a convention all db accessors accept a db as an optional last parameter in order to pass transaction contexts around

Reply:
We had that. Look for `AppContext`